lib/jwt/jwk/key_finder.rb

```diff
@@ -14,7 +15,8 @@ def initialize(options)
       end
 
       def key_for(kid)
-        raise ::JWT::DecodeError, 'No key id (kid) found from token headers' unless kid
+        raise ::JWT::DecodeError, 'No key id (kid) found from token headers' unless kid || @allow_nil_kid
+        raise ::JWT::DecodeError, 'Invalid type for kid header parameter' unless kid.nil? || kid.is_a?(String)
```

Comment:
I think checking just if the kid is a String is enough, `nil.is_a?(String) == false`
```suggestion
        raise ::JWT::DecodeError, 'Invalid type for kid header parameter' unless kid.is_a?(String)
```

Reply:
For my Ruby version (3.0.5), I get the following:

```ruby
p nil.is_a(String)
```

```
-:1:in `<main>': undefined method `is_a' for nil:NilClass (NoMethodError)
```

Even then, the whole point of this PR is to make this series of checks evaluate to `true` for the `unless` clauses in the presence of `nil`, so errors are not automatically raised if the application can handle them :)

Reply:
`is_a?` should be pretty basic functionality since the early days.

Reply:
Nvm, I have missed a `?` :sweat_smile:
Still, if it returns `false`, `unless` will not prevent the error from being raised.

Reply:
Youre right it needs to pass that line if it's nil. `unless` is so hard for the brain...